src/sdk/pynni/nni/curvefitting_assessor/modelfactory.py

```diff
@@ -0,0 +1,187 @@
+class CurveModel(object):
+    def __init__(self, target_pos):
+        self.target_pos = target_pos
+
+    def fit_theta(self):
+        '''use least squares to fit all default curves parameter seperately'''
+        x = range(1, self.point_num + 1)
+        y = self.trial_history
+        for i in range (NUM_OF_FUNCTIONS):
+            model = curve_combination_models[i]
+            try:
+                if model_para_num[model] == 2:
+                    a, b = optimize.curve_fit(all_models[model], x, y, maxfev = MAXFEV)[0]
+                    model_para[model][0] = a
+                    model_para[model][1] = b
+                elif model_para_num[model] == 3:
+                    a, b, c = optimize.curve_fit(all_models[model], x, y, maxfev = MAXFEV)[0]
+                    model_para[model][0] = a
+                    model_para[model][1] = b
+                    model_para[model][2] = c
+                elif model_para_num[model] == 4:
+                    a, b, c, d = optimize.curve_fit(all_models[model], x, y, maxfev = MAXFEV)[0]
+                    model_para[model][0] = a
+                    model_para[model][1] = b
+                    model_para[model][2] = c
+                    model_para[model][3] = d
+            except Exception as exception:
+                pass
+    
+    def filter_curve(self):
+        '''filter the poor performing curve'''
+        avg = np.sum(self.trial_history) / self.point_num
+        standard = avg * avg * self.point_num
+        predict_data = []
+        tmp_model = []
+        for i in range(NUM_OF_FUNCTIONS):
+            var = 0
+            model = curve_combination_models[i]
+            for j in range(1, self.point_num + 1):
+                y = self.predict_y(model, j)
+                var += (y - self.trial_history[j - 1]) * (y - self.trial_history[j - 1])
+            if var < standard:
+                predict_data.append(y)
+                tmp_model.append(curve_combination_models[i])
+        median = np.median(predict_data)
+        std = np.std(predict_data)
+        for model in tmp_model:
+            y = self.predict_y(model, self.target_pos)
+            if y < median + 3 * std and y > median - 3 * std:
+                self.effective_model.append(model)
+        self.effective_model_num = len(self.effective_model)
+
+    def predict_y(self, model, pos):
+        '''return the predict y of 'model' when epoch = pos'''
+        if model_para_num[model] == 2:
+            y = all_models[model](pos, model_para[model][0], model_para[model][1])
+        elif model_para_num[model] == 3:
+            y = all_models[model](pos, model_para[model][0], model_para[model][1], model_para[model][2])
+        elif model_para_num[model] == 4:
+            y = all_models[model](pos, model_para[model][0], model_para[model][1], model_para[model][2], model_para[model][3])
+        return y
+        
+    def f_comb(self, pos, sample):
+        '''return the value of the f_comb when epoch = pos'''
```

Comment:
it would be better to explain sample here.

Reply:
Add the explanation in MCMC_sampling function.